docs/source/status.rst

```diff
@@ -0,0 +1,199 @@
+| Float32/64        | ✓     | ✓     | ✓     | ✓          |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Decimal128        | ✓     | ✓     |       |            |       |       |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Date32/64         | ✓     | ✓     | ✓     | ✓          |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Time32/64         | ✓     | ✓     | ✓     | ✓          |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Timestamp         | ✓     | ✓     | ✓     | ✓          |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Duration          | ✓     | ✓     | ✓     |            |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Interval          | ✓     | ✓     | ✓     |            |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Fixed Size Binary | ✓     | ✓     | ✓     | ✓          |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Binary            | ✓     | ✓     | ✓     | ✓          |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Large Binary      | ✓     | ✓     | ✓     | ✓          |       |       |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Utf8              | ✓     | ✓     | ✓     | ✓          |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Large Utf8        | ✓     | ✓     | ✓     | ✓          |       |       |
++-------------------+-------+-------+-------+------------+-------+-------+
+
++-------------------+-------+-------+-------+------------+-------+-------+
+| Data type         | C++   | Java  | Go    | JavaScript | C#    | Rust  |
+| (nested)          |       |       |       |            |       |       |
++===================+=======+=======+=======+============+=======+=======+
+| Fixed Size List   | ✓     | ✓     | ✓     | ✓          |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| List              | ✓     | ✓     | ✓     | ✓          |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Large List        | ✓     |       |       |            |       |       |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Struct            | ✓     | ✓     | ✓     | ✓          |       | ✓     |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Map               | ✓     | ✓     |       | ✓          |       |       |
++-------------------+-------+-------+-------+------------+-------+-------+
+| Dense Union       | ✓     | ✓ (1) |       |            |       | ✓     |
```

Comment:
There are open PRs for Java Union and LargeList support. Would you prefer I update this doc when they are merged or mark as supported now? I am assuming this doc won't be live until 1.0 release.